nifi-nar-bundles/nifi-aws-bundle/nifi-aws-processors/src/main/java/org/apache/nifi/processors/aws/kinesis/stream/ConsumeKinesisStream.java

```diff
@@ -0,0 +1,708 @@
+        return context.getProperty(FAILOVER_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS);
+    }
+
+    private long getGracefulShutdownMillis(final PropertyContext context) {
+        return context.getProperty(GRACEFUL_SHUTDOWN_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS);
+    }
+
+    private long getCheckpointIntervalMillis(final PropertyContext context) {
+        return context.getProperty(CHECKPOINT_INTERVAL).asTimePeriod(TimeUnit.MILLISECONDS);
+    }
+
+    private int getNumRetries(final PropertyContext context) {
+        return context.getProperty(NUM_RETRIES).asInteger();
+    }
+
+    private long getRetryWaitMillis(final PropertyContext context) {
+        return context.getProperty(RETRY_WAIT).asTimePeriod(TimeUnit.MILLISECONDS);
+    }
+
+    private boolean isReportCloudWatchMetrics(final PropertyContext context) {
+        return context.getProperty(REPORT_CLOUDWATCH_METRICS).asBoolean();
+    }
+
+    private Optional<String> getKinesisEndpoint(final PropertyContext context) {
+        return context.getProperty(ENDPOINT_OVERRIDE).isSet()
+                ? Optional.of(StringUtils.trimToEmpty(context.getProperty(ENDPOINT_OVERRIDE).evaluateAttributeExpressions().getValue()))
+                : Optional.empty();
+    }
+
+    private Optional<String> getDynamoDBOverride(final PropertyContext context) {
+        return context.getProperty(DYNAMODB_ENDPOINT_OVERRIDE).isSet()
+                ? Optional.of(StringUtils.trimToEmpty(context.getProperty(DYNAMODB_ENDPOINT_OVERRIDE).evaluateAttributeExpressions().getValue()))
+                : Optional.empty();
+    }
+
+    private RecordReaderFactory getReaderFactory(final PropertyContext context) {
+        return context.getProperty(RECORD_READER).asControllerService(RecordReaderFactory.class);
+    }
+
+    private RecordSetWriterFactory getWriterFactory(final PropertyContext context) {
+        return context.getProperty(RECORD_WRITER).asControllerService(RecordSetWriterFactory.class);
+    }
+
+    private InitialPositionInStream getInitialPositionInStream(final PropertyContext context) {
+        return InitialPositionInStream.valueOf(StringUtils.trimToEmpty(context.getProperty(INITIAL_STREAM_POSITION).getValue()));
+    }
+
+    private DateTimeFormatter getDateTimeFormatter(final PropertyContext context) {
+        return DateTimeFormatter.ofPattern(context.getProperty(TIMESTAMP_FORMAT).evaluateAttributeExpressions().getValue());
+    }
+
+    private Date getStartStreamTimestamp(final PropertyContext context) {
+        return getStartStreamTimestamp(context, getDateTimeFormatter(context));
+    }
+
+    private Date getStartStreamTimestamp(final PropertyContext context, final DateTimeFormatter dateTimeFormatter) {
+        final String streamTimestamp = context.getProperty(STREAM_POSITION_TIMESTAMP).getValue();
+        return new Date(LocalDateTime.parse(streamTimestamp, dateTimeFormatter).toInstant(ZoneOffset.UTC).toEpochMilli());
```

Comment:
"Stream Position Timestamp" property is expected to be configured in UTC now, which is not really straightforward on the UI because the "aws.kinesis.approximate.arrival.timestamp" FF attribute is written in local TZ (NiFi server's system TZ).
I think local system TZ should be used for this property too (and convert it to UTC for Kinesis in the code) or it should be documented in the property's description that UTC is expected.

Reply:
It works properly now with local system time zone. Thanks

A minor note: `withZoneSameInstant(ZoneOffset.UTC)` is not necessarily needed, because in the end we need the Instant and this method does not change the Instant (and for this reason it does not cause any problem either).

Reply:
I think I'll leave this as-is because it's a bit more explicit what's going on